### PR TITLE
Add support for YPbPr output.

### PR DESCRIPTION
**Pull request by sorgelig**
Hi,
please consider to add YPbPr video output. Relevant setting in INI file is already included in latest FW. All my cores are already support YPbPr. Please see the info here: https://github.com/mist-devel/mist-board/wiki/YPbPr_Cable

### CODE REVIEW

**Comment by sorgelig** [comment]
I'm trying to understand why nag screen has positive hsync/vsync pulses instead on normal negative pulses. But code is too confusing. Is it possible to make VSync/HSync pulses negative during nag screen?

File: fpga/mist/minimig_mist.qsf [hunk 1]
```diff
@@ -337,6 +337,7 @@ set_instance_assignment -name CURRENT_STRENGTH_NEW 8MA -to VGA_VS
 
 
 
+set_global_assignment -name SYSTEMVERILOG_FILE ../../rtl/minimig/video_mixer.sv
```
Comment (rkrajnc) [inline fpga/mist/minimig_mist.qsf:340]: It would be nice if this was just plain Verilog, so it would be compatible with all simulators, where SV support is pretty weak. I guess just the packed array needs rewrite.

File: rtl/minimig/amber.v [hunk 2]
```diff
@@ -344,15 +345,12 @@ wire [  8-1:0] bm_b;
 wire           bm_osd_blank;
 wire           bm_osd_pixel;
 
-assign bm_hsync     = dblscan ? sd_lbuf_o_d[29] : varbeamen ? _hsync_in : ns_csync;
```
Comment (rkrajnc) [inline rtl/minimig/amber.v:347]: Is is really necessary to remove this two lines? Ideally, the code should still work with the added yuv encoder or without. Any simple way to avoid this?

File: rtl/minimig/video_mixer.sv [hunk 3]
```diff
@@ -0,0 +1,98 @@
+`timescale 1ns / 1ps
```
Comment (rkrajnc) [inline rtl/minimig/video_mixer.sv:1]: Please, no timescale in non-sim, non-toplevel files.

File: rtl/minimig/video_mixer.sv [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+wire [18:0] pr_8 = 19'd32768 + ({red, 6'd0} + {red, 5'd0} + {red, 4'd0}) - ({green, 6'd0} + {green, 4'd0} + {green, 3'd0} + {green, 2'd0} + {green, 1'd0}) - ({blue, 4'd0} + {blue , 1'd0});
+
+wire [7:0] y  = ( y_8[17:8] < 16) ? 8'd16 : ( y_8[17:8] > 235) ? 8'd235 :  y_8[15:8];
+wire [7:0] pb = (pb_8[17:8] < 16) ? 8'd16 : (pb_8[17:8] > 240) ? 8'd240 : pb_8[15:8];
+wire [7:0] pr = (pr_8[17:8] < 16) ? 8'd16 : (pr_8[17:8] > 240) ? 8'd240 : pr_8[15:8];
+
+assign VGA_R  = ypbpr ? (ypbpr_full ? yuv_full[pr-8'd16] : pr[7:2]) :   red[7:2];
+assign VGA_G  = ypbpr ? (ypbpr_full ? yuv_full[y -8'd16] :  y[7:2]) : green[7:2];
+assign VGA_B  = ypbpr ? (ypbpr_full ? yuv_full[pb-8'd16] : pb[7:2]) :  blue[7:2];
+assign VGA_VS = (scandoubler_disable | ypbpr) ? 1'b1 : ~vsync_p;
```
Comment (rkrajnc) [inline rtl/minimig/video_mixer.sv:95]: If I read this properly, you are inverting the hsync and vsync from the minimig core? Why is this necessary? The existing polarity has proven to work just fine, and while I know that this might not be problematic in most cases, it could be in others. I'm reluctant to accept this change, since minimig has worked just fine for everybody until now.

**Review by rkrajnc — CHANGES_REQUESTED** [review]
Hi, thank you for the pull request, this certainly come in handy for some people.

Unfortunately, I have no way of testing this, since I have no TV/monitor that would accept YPbPr input. I guess the changes were tested by you?

I added comments in the code where applicable, and there are two additional issues:
- no binary files in the repository, so please remove the core file (better to place it in mist binaries repo anyway).
- if at all possible, please redo your changes on the dev branch instead of master. I use dev for all my work, and merge to master only release-ready code. Because of some mistakes I made a while back, I wouldn't be able to merge the two branches together cleanly. If it's OK with you, I could still do the pull manually, but edit history would be lost. DEV branch is preferred ;)

Thanks!

**Comment by sorgelig** [comment]
I'm not really familiar with pull requests. Original idea was just show the changes for you so you could extract needed changes. Binary actually is temporary and i planned to remove it when you would add changes into main branch. Anyway. I will re-do the changes according to your requirements.
